File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        int n = intervals.size();
        vector<vector<int>> output;
        sort(intervals.begin(),intervals.end());
        for(auto interval: intervals){
            if(output.empty() || output.back()[1] < interval[0])
                output.push_back(interval);
            else 
                output.back()[1] = max(output.back()[1],interval[1]);
        }
        return output;
    } 
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:

        ans=[]
        intervals.sort(key=lambda x:(x[0],x[1]))

        curr=intervals[0]
        

        for i in range(1,len(intervals)):
            nextt=intervals[i]
            if curr[1]>=nextt[0]:
                curr=[curr[0],max(curr[1],nextt[1])]
                
            else:
                ans.append(curr)
                curr=nextt
        ans.append(curr)

        return ans
        